0019/README.md

```diff
@@ -152,6 +152,122 @@ Endpoint metadata get response message payload MUST be an Avro-encoded object wi
+```
+
+Endpoint metadata keys get request message payload MUST be an [Avro-encoded](https://avro.apache.org/) object with the following schema ([0019-endpoint-metadata-get-keys-request.avsc](./0019-endpoint-metadata-keys-get-request.avsc)):
+
+```json
+{
+    "namespace":"org.kaaproject.ipc.epmmp.gen.v1",
+    "name":"EndpointMetadataKeysGetRequest",
+    "type":"record",
+    "doc":"Interservice endpoint metadata keys get request",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds since the timestamp until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"endpointId",
+            "type": "string",
+            "doc":"Identifier of the endpoint, on behalf of which metadata is requested"
+        }
+    ]
+}
+```
+
+
+## Endpoint metadata response
+
+*Endpoint metadata response* message MUST be sent by the repository in response to [Endpoint metadata keys get request message](#endpoint-metadata-keys-get-request).
+Repository MUST publish endpoint metadata keys get response message to the subject provided in the NATS `replyTo` field of the request.
+
+Endpoint metadata keys get response message payload MUST be an Avro-encoded object with the following schema ([0019-endpoint-metadata-keys-get-response.avsc](./0019-endpoint-metadata-keys-get-response.avsc)):
+
+```json
+{
+     "namespace":"org.kaaproject.ipc.epmmp.gen.v1",
+     "name":"EndpointMetadataKeysGetResponse",
+     "type":"record",
+     "doc":"Endpoint metadata keys get response",
+     "fields":[
+         {
+             "name":"correlationId",
+             "type":"string",
+             "doc":"Message id primarily used to track message processing across services"
+         },
+         {
+             "name":"timestamp",
+             "type":"long",
+             "doc":"Message creation unix timestamp in milliseconds"
+         },
+         {
+             "name":"timeout",
+             "type":"long",
+             "default":0,
+             "doc":"Amount of milliseconds since the timestamp until the message expires. Value of 0 is reserved to indicate no expiration."
+         },
+         {
+             "name":"endpointId",
+             "type": "string",
+             "doc":"Identifier of the endpoint, on behalf of which metadata is requested"
+         },
+         {
+             "name":"keys",
+             "type":{
+               "type":"array",
+               "items":"string"
+             },
+             "default":[],
+             "doc":"List of metadata fields"
```

Comment:
I thought that these events is about EP metadata keys, not values?

If so, then update the doc:
"doc":"List of the endpoint metadata keys"